base_partner_sequence/models/partner.py

```diff
@@ -37,7 +37,7 @@ def write(self, vals):
                 and not partner.ref
             ):
                 partner_vals["ref"] = partner._get_next_ref(vals=partner_vals)
-            super(ResPartner, partner).write(partner_vals)
+            super().write(partner_vals)
```

Comment:
it must be `super(ResPartner, partner).write(partner_vals)` because this line is in the loop `for partner in self:`